File: Objc/Search/LinearSearch/LinearSearch.m
//
//  LinearSearch.m
//  Algorithm
//
//  Created by Kuan-Wei Lin on 3/8/17.
//  Copyright © 2017 TaiwanMobile. All rights reserved.
//

#import "LinearSearch.h"

@implementation LinearSearch

- (void)nonSentinelLinearSearch:(NSArray *)array key:(NSNumber *)key
{
    NSMutableArray *data = [array mutableCopy];
    
    int index = 0;
    
    while (index < data.count)
    {
        if (data[index++] == key)
        {
            int result = index - 1;
            NSLog(@"result = %i", result);
        }
        else
        {
            NSLog(@"search result not found");
        }
    }
}

- (void)sentinelLinearSearch:(NSArray *)array key:(NSNumber *)key
{
    NSMutableArray *data = [array mutableCopy];
    
    int index = 0;
    
    [data addObject:key];
    
    while (data[index++] != key)
    {
        NSLog(@"index = %i", index);
    }
    
    [data removeLastObject];
    
    int result = (--index == data.count) ? -1 : index;
    
    NSLog(@"result = %i", result);
}


@end

File: Swift/Search/LinearSearch.swift
//
//  LinearSearch.swift
//  Algorithm_Swift
//
//  Created by Kuan-Wei Lin on 3/16/17.
//  Copyright © 2017 cracktheterm. All rights reserved.
//

import UIKit

class LinearSearch: NSObject {
    
    func sentinelLinearSearch(array:[Int], key:Int)
    {
        var index = 0;
        
        var data = array;
        
        data.append(key);
        
        while array[index] != key
        {
            print("index = \(index)")
            
            index += 1;
        }
        
        data.removeLast();
        
        let result = (index == data.count) ? -1 : index;
        print("result = \(result)");
    }
    
    func nonSentinelLinearSearch(array:[Int], key:Int)
    {
        var index = 0;
        
        while index < array.count
        {
            
            if array[index] == key
            {
                let result = index;
                print("result = \(result)");
            }
            else
            {
                print("search result not found");
            }
            
            index += 1
        }
    }
}
